File: leetcode/2872-largest-element-in-an-array-after-merge-operations/solution.c
long long max(long long a, long long b) {
    return a>b?a:b;
}
long long maxArrayValue(int* nums, int n){
    long long max_sum = 0;
    long long sum = 0;
    for(int i = n-1;i>=0;i--) {
        if(nums[i] > sum) {
            max_sum = max(sum, max_sum);
            sum = nums[i];
        } else {
            sum += nums[i];
        }
    }
    return max(max_sum, sum);
}

File: leetcode/2872-largest-element-in-an-array-after-merge-operations/solution.rs
impl Solution {
    pub fn max_array_value(nums: Vec<i32>) -> i64 {
        use std::cmp::max;
        let mut max_sum = 0;
        let mut sum = 0;
        for x in nums.into_iter().rev().map(|x| x as i64) {
            if x > sum {
                max_sum = max(max_sum, sum);
                sum = x;
            } else {
                sum += x;
            }
        }
        return max(max_sum, sum);
    }
}
